enterprise/micronaut/src/org/netbeans/modules/micronaut/symbol/MicronautSymbolFinder.java

```diff
@@ -111,6 +112,11 @@ public final class MicronautSymbolFinder extends EmbeddingIndexer implements Pro
 
     private final Map<FileObject, Boolean> map = new WeakHashMap<>();
     private final Map<ClasspathInfo, List<ClassSymbolLocation>> cache = new WeakHashMap<>();
+    private final ChangeListener listener = (evt) -> {
+        if (cache.keySet().contains(evt.getSource())) {
+            cache.clear();
+        }
+    };
```

Comment:
nitpick: better name would be `cacheCleaner` or something similar. Or it could be inlined so that it doesn't need a name.